cmake/variables.cmake

```diff
@@ -156,7 +156,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
   list(APPEND ML_COMPILE_DEFINITIONS BOOST_ALL_NO_LIB)
 endif()
 
-# Dictate which flags to use for a "Release" build
+# Dictate which flags to use for "Release" and "Debug" builds
```

Comment:
At the moment we unconditionally add `-g` to these flags.

That was based on our earlier discovery that CMake's default `RelWithDebInfo` flags were pretty suboptimal.

But our primary builds really are "releases with debug info".

I don't think you should change anything more in this PR, but if this one goes through successfully then in a followup we should probably set an extra variable here for each platform to define our preferred `RelWithDebInfo` flags, take `-g` out of the unconditional flags, and switch our CI builds to do `RelWithDebInfo` builds. Then we'd be using CMake targets in the intended way.

But, like I said, don't try to do it all in this PR. Let's go one step at a time in case there's some unforeseen side effect of this one.